mne_bids/write.py

```diff
@@ -1161,13 +1169,6 @@ def write_raw_bids(raw, bids_basename, output_path, events_data=None,
                               'overwrite to True.' % bids_fname)
     _mkdir_p(os.path.dirname(bids_fname))
 
-    convert = ext not in ALLOWED_EXTENSIONS[kind]
-
-    if kind == 'meg' and convert and not anonymize:
-        raise ValueError('Got file extension %s for MEG data, ' +
-                         'expected one of %s' %
-                         ALLOWED_EXTENSIONS['meg'])
```

Comment:
This scares me a bit. We should discourage people from storing files in whatever format they wish to. If people want to store in a new format, the case should be made in the BIDS specification first. More formats is a burden for software developers. For most use cases, I would argue that conversion + sidecars will be a lossless conversion.

Reply:
I would argue that this would actually make it much easier to use files that are not BIDS standard yet. I think that error needs to be added back in but that is actually related to this PR but not a problem stemming from the PR: if you have a file format that is not recognized (.easy for instance) there is no reader and so this line in write_raw_bids fails 

```
raw_orig = reader[ext](**raw._init_kwargs)
```
with a cryptic not in dictionary error. Thus, no one would ever get to that error anyway because all MEG formats that have readers are supported by BIDS. This just adds the option that if have tested it thoroughly and know that the SET or BTI or other format is functionally equivalent to FIF and you want to move toward more standardization, you can do so. On the other hand, if you have a FIF eeg-only file for instance and for whatever reason the conversion to BV is lossy, yes you should submit a PR and fix it, but just practically some people aren't going to have the time/ability to do that and if that isn't supported then someone might not be able to use BIDS, which could seriously hamper their project if their tools depend on BIDS like it seems is the way future development is headed. Hence the not converting idea is purely pragmatic and I was just picturing it being used until any problem is fixed.

Reply:
Or rather the error was caught on line 1043

```
if ext not in [this_ext for data_type in ALLOWED_EXTENSIONS
                   for this_ext in ALLOWED_EXTENSIONS[data_type]]:
        raise ValueError('Unrecognized file format %s' % ext)
```

Reply:
Also I would add I have EDF files that I would like to store in BV because I think at some point a preferred file type will emerge and everyone will start moving toward that as conversions are shown to be stable.